### PR TITLE
fix: trailing slash shouldn't fail the path matching

### PR DESCRIPTION
**Pull request by hansyulian**
discussion:
- https://github.com/TanStack/router/discussions/1364

### CODE REVIEW

File: packages/react-router/src/link.tsx [hunk 1]
```diff
@@ -320,35 +320,35 @@ export type LinkOptions<
...
+  ? ToRest extends ['/']
+  ? Join<['/', ...FromRest, '/']>
+  : ResolveRelativePath<Join<FromRest>, Join<ToRest>>
+  : never
+  : Split<TTo> extends ['.', ...infer ToRest]
+  ? ToRest extends ['/']
+  ? Join<[TFrom, '/']>
+  : ResolveRelativePath<TFrom, Join<ToRest>>
+  : CleanPath<Join<['/', ...Split<TFrom>, ...Split<TTo>]>>
+  : never
```
Comment (SeanCassiere) [inline packages/react-router/src/link.tsx:351]: This shouldn't be getting flattened like this. Could you check if your prettier set-up is correctly picking up the prettier config at the root.

**Review by SeanCassiere — CHANGES_REQUESTED** [review]
Just prettier fixes for me.

File: packages/react-router/src/link.tsx (hunk 1 above)
Comment (hansyulian) [reply packages/react-router/src/link.tsx]: hi @SeanCassiere thx for the feedback, have fixed the prettier issue

**Review by SeanCassiere — APPROVED** [review]
LGTM

**Comment by mudasirpandith** [comment]
@SeanCassiere i am getting same issue, i am having `v1.28.1`

**Comment by TkDodo** [comment]
@SeanCassiere @mudasirpandith I observed this as well with `trailingSlash:always` for sub-routes. Reproduction would be this:

https://stackblitz.com/edit/tanstack-router-kh6hmr

and I think this PR should fix it:
- https://github.com/TanStack/router/pull/1711

**Comment by mudasirpandith** [comment]
@TkDodo What should be my approach to overcome this in my project?

**Pull request merged**